src/grants/GrantCheck.ts

```diff
@@ -0,0 +1,176 @@
+import { Logger } from "matrix-appservice-bridge";
```

Comment:
Dropping into a thread (@justinbot):

>  (1) how much boilerplate each connection seems to require 

So from my side I've tried to reduce the boilerplate as much as I can, but each connection sort of needs to do different things in subtle but annoying ways.  I originally tried to make this part of the ConnectionManager although it was difficult due to each requirement, so it ended up per connection with as much logic as I could manage abstracted into the grantChecker class. 

Of course, I've been looking at this for so long there might be another way.